kernel/kernel-api/src/main/java/io/delta/kernel/internal/InternalScanFileUtils.java

```diff
@@ -54,8 +54,13 @@ private InternalScanFileUtils() {}
         TABLE_ROOT_DATA_TYPE,
         false /* nullable */);
 
+    // TODO what to do about these indices? can we do this differently?
```

Comment:
The schema here is the same as the schema of the returned scan files. That shouldn't change depending on the stats or not. If the scan file schema changes, these APIs should also change.

The constants in LogReplay may not in sync with these as LogReplay can select different columns and drop them.

Reply:
For now we are returning the stats as part of the returned scan files. It doesn't affect the other indices since it's at the end but this `SCAN_FILE_SCHEMA` isn't always completely true if the stats are read